serde-generate/runtime/golang/src/lcs/deserializer.go

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) Facebook, Inc. and its affiliates
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+package lcs
+
+import (
+	"bytes"
+	"errors"
+
+	"github.com/facebookincubator/serde-reflection/serde"
+)
+
+// MaxSequenceLength is max length allowed for sequence
+const MaxSequenceLength = (1 << 31) - 1
+
+const maxUint32 = uint64(^uint32(0))
+
+// Deserializer implements `serde.Deserializer` interface for deserializing LCS serialized bytes
+type Deserializer struct {
+	buf *bytes.Buffer
+}
+
+// NewDeserializer creates a new `serde.Deserializer`
+func NewDeserializer(input []byte) serde.Deserializer {
+	return &Deserializer{buf: bytes.NewBuffer(input)}
+}
+
+func (d *Deserializer) DeserializeBytes() ([]byte, error) {
+	len, err := d.DeserializeLen()
+	if err != nil {
+		return nil, err
+	}
+	ret := make([]byte, len)
+	_, err = d.buf.Read(ret)
+	return ret, err
+}
+
+func (d *Deserializer) DeserializeLen() (int, error) {
```

Comment:
`uint32` makes sense for LCS but in general (bincode) it should be `uint64` (my bad). Java was an exception because it has no signed integers and arrays are limited to 2^31 elements.